UITests/UITests.App/TestInterop.cs

```diff
@@ -43,7 +43,7 @@ private static void LogMessage(string level, string format, object[] args)
 
             // Send back to Test Harness via AppService
             // TODO: Make this a cleaner connection/pattern
-            _ = ((App)Application.Current).SendLogMessage(level, message);
+            await ((App)Application.Current).SendLogMessage(level, message).ConfigureAwait(false);
```

Comment:
I don't understand this change. The PR title says that this is just about suppressing warnings, yet this is introducing a functionality behavior, with no explanation. What was the warning appearing before on this line?

Reply:
CS4014

Reply:
I think the point @RosarioPulella was calling out here is that since we don't care about status there's no value to having this be `async void` (which is usually used only for UI methods being called from event handlers from the UI thread).

If we care about the status and want to do something actionable than we should return a `Task`, but since we don't care about this at all and are forwarding this on to another place doing the logging, I don't expect any issues here unless something catastrophic happens.

Was just having the discard that's there causing a warning? If so, what was it? Just want to understand the change in the pattern.

Thanks!

Reply:
Yes, the discard is what's causing the warning. Awaiting the method fixes the issue. But to await this, we need to make this method async but disable the context return to not hit any async issues that commonly arise from this pattern. Hence the changes!!

I might be wrong or misunderstood here but this is exactly what I did when I did hit a similar issue.

Reply:
> "the discard is what's causing the warning."

In that case we should just disable that warning, given that we're using discards in other places too.

> "Awaiting the method fixes the issue."

Yes, which is also a functional change, which is not what this PR is supposed to be. Of course, doing functional changes is fine, but they need to be justified, discussed in an issue, detailed in the PR, and tested.

> "disable the context return to not hit any async issues that commonly arise from this pattern"

That doesn't sound right at all, can you elaborate on what you mean there? Setting `ConfigureAwait(false)` simply uses a configured awaiter that modifies the default behavior by not capturing the original synchronization context, which means that the continuation is free to be scheduled on whatever thread might be available at that time, which has less overhead. It is therefore mostly an optimization that can be done when the specific synchronization context the code is running on doesn't matter, but it's not meant to "fix async issues" (what do you even mean by that?). Not to mention that change can't possibly matter here since there is no code at all in the continuation anyway - that await is the last statement in the whole method. The only observable change would be what thread ends up throwing the exception in case that operation fails (which it shouldn't anyway). Additionally, this change is adding an unnecessary async state machine in this method that's never actually needed, since callers wouldn't be able to await the whole operation anyway as the method just returns `void`.

The way I see it, we have two options here:
- Just leave the code as it was (which was fine)
- Make the code return `Task` and directly forward the returned one from that call, without async state machine. Then individual callers would be free to eiter explicitly await, or discard the task on their end. This would likely need more changes at each callsite and it would be of dubious value given that as of right now, no caller is ever awaiting this method anyway and we never feel the need to do so.

Personally I would:
- Disable the discard warning in the whole solution
- Add a discard as suggested in the other change above
- Discard this change here entirely

Reply:
I see, it worked for me as I've used this pattern without any issues till now.

But now I understand clearly as I've also checked the decompiled souces to see what was generated before and after my change.

Let's ignore the warning solution wide. Thanks.

Reply:
Awesome, thanks! 😄